src/integration.md

```diff
@@ -0,0 +1,569 @@
+rules can then be passed to `BilinearFormIntegrator` or
+`LinearFormIntegrator` objects (using custom integration rules with
+mixed meshes currently requires specialized handling).
+
+## Coordinate Transformations
+
+The coordinate transformation from the reference element to an
+individual mesh element is performed by the `ElementTransformation`
+class. Objects of this class are prepared by the `Mesh` object and
+retrieved in various ways depending on context.
+
+For standard mesh elements
+
+```c++
+for (int e=0; e<mesh->GetNE(); e++)
+{
+   ElementTransformation *Trans = mesh->GetElementTransformation(e);
+
+   ...
+}
+```
+
+or for boundary elements
+
+```c++
+for (int be=0; be<mesh->GetNBE(); be++)
+{
+   ElementTransformation *Trans = mesh->GetBdrElementTransformation(be);
+
+   ...
+}
+```
+
+or for faces (usually in a Discontinuous Galerkin (DG) context)
+
+```c++
+for (int f=0; f<mesh->GetNumFaces(); f++)
+{
+   FaceElementTransformation *FETrans = mesh->GetFaceElementTransformation(f);
+
+   ...
+}
+```
+
+or, finally, for boundary faces in a DG context
+
+```c++
+for (int bf=0; bf<mesh->GetNBE(); bf++)
+{
+   FaceElementTransformation *FETrans = mesh->GetBdrFaceElementTransformation(bf);
+
+   ...
+}
+```
+
+A `FaceElementTransformation` object is a convenience object for
+easily accessing the three `ElementTransformation` objects associated
+with a mesh face and its two neighboring elements. In the case of
+boundary faces one of the neighboring element transformation objects
+is not present.
+
+In addition to transforming coordinates between the reference and
+global coordinate systems an `ElementTransformation` object can be
+used to compute the following quantities related to the Jacobian
+matrix:
+
+| Name                 | C++ Expression                                       | Formula |
+|----------------------|------------------------------------------------------|---------|
+| Jacobian Matrix      | `const DenseMatrix &J = Trans.Jacobian()`            | $\{\bf J}_\{ij} = \frac\{\partial x_i}\{\partial u_j}$ |
+| Jacobian Determinant | `double detJ = Trans.Weight()`                       | $\|\{\bf J}\|$ |
+| Inverse Jacobian     | `const DenseMatrix &InvJ = Trans.InverseJacobian()`  | $\{\bf J}^\{-1}$ |
+| Adjugate Jacobian    | `const DenseMatrix &AdjJ = Trans.AdjugateJacobian()` | $\|\{\bf J}\|\,\{\bf J}^\{-1}$ |
+
+Since these quantities can be expensive to compute the
+`ElementTransformation` object will avoid recomputing values whenever
+possible. However, once a new quadrature point is set, using
+`ElementTransformation::SetIntPoint()`, any cached values will be
+overwritten by subsequent calls to the above functions.
+
+## Writing Custom Integrators
+
+Element-wise integration arises in various places in the finite element
+method. A few of the most common occurrences are square and rectangular
+bilinear form operators, linear form operators, and the calculation of
```

Comment:
Asa Tzanio caught below linearForms don't produce or represent operators, but instead they produce finite dimensional vectors and represent infinite dimensional vectors/functions.  I'm not sure quite what to name the thing in the first column though because vector is confusing in this context and so is function.  Naming things is hard...